nav2_map_server/test/component/CMakeLists.txt

```diff
@@ -1,54 +1,113 @@
 include_directories(${PROJECT_SOURCE_DIR}/test)
 
 # map_server component test
-ament_add_gtest_executable(test_map_server_node
-  test_map_server_node.cpp
+ament_add_gtest_executable(test_map_server_2d_node
+    test_map_server_2d_node.cpp
   ${PROJECT_SOURCE_DIR}/test/test_constants.cpp
 )
-ament_target_dependencies(test_map_server_node rclcpp nav_msgs)
-target_link_libraries(test_map_server_node
+ament_target_dependencies(test_map_server_2d_node rclcpp nav_msgs nav2_msgs nav2_util)
+target_link_libraries(test_map_server_2d_node
+    ${map_io_library_name}_2d
+    ${library_name}
+  stdc++fs
+)
+
+ament_add_test(test_map_server_2d_node
+  GENERATE_RESULT_FOR_RETURN_CODE_ZERO
+  COMMAND "${CMAKE_CURRENT_SOURCE_DIR}/test_map_server_2d_launch.py"
+  WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"
+  ENV
+    TEST_DIR=${TEST_DIR}
+    TEST_LAUNCH_DIR=${TEST_LAUNCH_DIR}
+    TEST_EXECUTABLE=$<TARGET_FILE:test_map_server_2d_node>
+)
+
+# map_server_3d component test
+ament_add_gtest_executable(test_map_server_3d_node
+    test_map_server_3d_node.cpp
+  ${PROJECT_SOURCE_DIR}/test/test_constants.cpp
+)
+ament_target_dependencies(test_map_server_3d_node rclcpp nav2_msgs
+  nav2_util sensor_msgs geometry_msgs)
+target_link_libraries(test_map_server_3d_node
+  ${map_io_library_name}_3d
```

Comment:
The same: why do we need to link a test over MapIO library?

Reply:
Removed them.